src/FileReader_Timefilter.py

```diff
@@ -0,0 +1,99 @@
+
+    def inspect_group(self, item, name_prefix=''):
+        if isinstance(item, h5py.Dataset):
+            # add --> spezifische Schlüsselnamen prüfen, die auf Zeitstempel hinweisen
+            if 'time' in name_prefix.lower() or 'timestamp' in name_prefix.lower():
+                print(f"Zeitstempel gefunden in: {name_prefix}, Daten: {item[:]}")
+
+                # Hier wird für jedes Element in item geprüft und konvertiert
+                for ts in item[:]:
+
+                    try:
+                        # Versuche, als Unix-Zeitstempel zu interpretieren
+                        ts_converted = datetime.fromtimestamp(int(ts), timezone.utc)
+                    except (ValueError, TypeError):
+
+                        # Falls das fehlschlägt, verarbeite als ISO-Format String
+                        try:
+                            ts_str = ts.decode('utf-8')  # Funktioniert nur, wenn ts ein Bytes-Objekt ist
+                            ts_converted = parser.parse(ts_str)
+                        except (ValueError, TypeError, AttributeError):
+                            print(f"Unlesbarer Zeitstempel: {ts} in {item.name}")
+                            continue
+
+                        try:
+                            ts_converted = parser.parse(ts_str)
+                        except (ValueError, TypeError, AttributeError):
+                            print(f"Unlesbarer Zeitstempel: {ts} in {item.name}")
+                            continue
+
+                        decade = (ts_converted.year // 10) * 10
+                        if decade not in self.timestamps_by_decade:
+                            self.timestamps_by_decade[decade] = []
+
+                        self.timestamps_by_decade[decade].append(ts_converted)
+
+                    self.all_timestamps.append(ts_converted)
+
+        elif isinstance(item, h5py.Group):
+            for name in item:
+                self.inspect_group(item[name], name_prefix=f"{name_prefix}/{name}")
+
+    def visualize_timestamps(self):
+        dates = [dt.replace(tzinfo=None) for dt in self.all_timestamps]
+
+        # Einfacher Plot: Datum gegen Index
+        plt.figure(figsize=(10, 6))
+        plt.plot(dates, np.arange(len(dates)), marker='o', linestyle='-')
+        plt.xlabel('Datum')
+        plt.ylabel('Zeitstempel Index')
+        plt.title('Visualisierung der gefundenen Zeitstempel')
+        plt.xticks(rotation=45)
+        plt.tight_layout()
+        plt.show()
+
+    def write_timestamps_to_csv(self, filepath):
+        with open(filepath, 'w', newline='') as file:
+            writer = csv.writer(file)
+            writer.writerow(['Index', 'Datum', 'Jahrzehnt'])
+
+            for decade, timestamps in sorted(self.timestamps_by_decade.items()):
+                for index, dt in enumerate(timestamps):
+                    dt_str = dt.strftime('%Y-%m-%d %H:%M:%S')
+                    writer.writerow([index, dt_str, str(decade)])
+
+
+# Nutzung:
```

Comment:
Am besten rufst du so etwas in einem Controller aus, z.B. main.py. Sonst hat keiner mehr einen Pla was genau im Programm Phase ist.